app/src/main/java/org/oppia/android/app/application/ApplicationComponent.kt

```diff
@@ -14,6 +42,41 @@ import javax.inject.Provider
  * This component will be subclasses for specific contexts (such as test builds, or specific build
  * flavors of the app).
  */
+@Singleton
+@Component(
+  modules = [
+    ApplicationModule::class, DispatcherModule::class,
+    LoggerModule::class, OppiaClockModule::class,
+    ContinueModule::class, FractionInputModule::class,
+    ItemSelectionInputModule::class, MultipleChoiceInputModule::class,
+    NumberWithUnitsRuleModule::class, NumericInputRuleModule::class,
+    TextInputRuleModule::class, DragDropSortInputModule::class,
+    InteractionsModule::class, GcsResourceModule::class,
+    GlideImageLoaderModule::class, ImageParsingModule::class,
+    HtmlParserEntityTypeModule::class, CachingModule::class,
+    QuestionModule::class, DebugLogReportingModule::class,
+    AccessibilityProdModule::class, ImageClickInputModule::class,
+    LogStorageModule::class, IntentFactoryShimModule::class,
+    ViewBindingShimModule::class, PrimeTopicAssetsControllerModule::class,
+    ExpirationMetaDataRetrieverModule::class, RatioInputModule::class,
+    UncaughtExceptionLoggerModule::class, ApplicationStartupListenerModule::class,
+    LogUploadWorkerModule::class, WorkManagerConfigurationModule::class,
+    HintsAndSolutionConfigModule::class, HintsAndSolutionDebugModule::class,
+    FirebaseLogUploaderModule::class, NetworkModule::class, PracticeTabModule::class,
+    PlatformParameterModule::class, PlatformParameterSingletonModule::class,
+    ExplorationStorageModule::class, DeveloperOptionsStarterModule::class,
+    DeveloperOptionsModule::class, PlatformParameterSyncUpWorkerModule::class,
+    NetworkConnectionUtilDebugModule::class, NetworkConfigProdModule::class, AssetModule::class,
+    LocaleProdModule::class, ActivityRecreatorProdModule::class, ActivityRouterModule::class,
+    NumericExpressionInputModule::class, AlgebraicExpressionInputModule::class,
+    MathEquationInputModule::class, SplitScreenInteractionModule::class,
+    LoggingIdentifierModule::class, ApplicationLifecycleModule::class,
+    // TODO(#59): Remove this module once we completely migrate to Bazel from Gradle as we can then
+    //  directly exclude debug files from the build and thus won't be requiring this module.
+    NetworkConnectionDebugUtilModule::class, LoggingIdentifierModule::class, SyncStatusModule::class
```

Comment:
@vrajdesai78 Why are all the modules added here now?
I think you are referencing the previous PR incorrectly. All these modules must have been transferred somewhere and at that new place you might need to add your newly created module. First check how the current develop dagger architecture is and based on that make changes.

Reply:
Shifted to DeveloperApplicatoinComponent from ApplicationComponent as per latest develop.